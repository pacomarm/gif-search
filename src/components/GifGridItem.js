import React from 'react'

import Card from 'react-bootstrap/Card';
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {
    return (
        // <div className="card animate__animated animate__jello">
        //     <img src={url} alt={title}/>
        //     <p> {title} </p>
        // </div>
        <Card className="cardd">
            <Card.Img id="im" className="pic" variant="top" src={url} alt={title} />
            <Card.Body className="hidden-mobile">
                <Card.Title id="tit">{title}</Card.Title>
            </Card.Body>
        </Card>
    )
}

GifGridItem.propTypes = {

    title : PropTypes.string.isRequired,
    url : PropTypes.string.isRequired,

}
