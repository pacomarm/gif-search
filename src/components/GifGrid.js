import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect( () => {
        getGifs(category)
            .then(imgs=>setImages(imgs));
    }, [category])

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                <ol>
                    {
                        images.map(gif => (
                            <GifGridItem 
                                key={gif.id} 
                                {...gif}
                            />
                        ))
                    }
                </ol>
            </div>
        </>
    )
}
