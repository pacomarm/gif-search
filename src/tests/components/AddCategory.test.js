import React from 'react'
import { shallow } from 'enzyme'
import {AddCategory} from '../../components/AddCategory'
import { ToggleButton } from 'react-bootstrap';
import '@testing-library/jest-dom'

describe('Tests in AddCategory component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories}/> );

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories}/> );
    })

    test('should show component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change text box', () => {

        const input = wrapper.find('input');
        const value = 'Waddup'
        input.simulate('change', {target:{value}}); 
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('should not submit info on submit', () => {

        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled(); 
    })
    
})
