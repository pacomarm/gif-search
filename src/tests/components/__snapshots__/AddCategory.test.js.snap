// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in AddCategory component should show component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
